model/src/main/java/org/hibernate/performance/search/model/application/DomainDataUpdater.java

```diff
@@ -120,7 +118,7 @@ public void updateQuestionnaire(Session session, int invocation, int questionnai
 		session.merge( definition );
 
 		List<QuestionnaireInstance> instances = new EmployeeRepository( session ).findByDefinition( definition );
-		for ( QuestionnaireInstance instance : instances ) {
+		for ( QuestionnaireInstance instance : choose( instances, 3 ) ) {
```

Comment:
Wouldn't it be more reasonable to include this filtering in the `findByDefinition` query above? From what I understand you may very well be loading hundreds of Questionnaire instances, only to use 2.

Reply:
I agree, I'm changing it...